src/sema/expression.rs

```diff
@@ -7861,16 +7874,22 @@ pub fn call_expr(
         {
             new(loc, ty, args, context, ns, symtable, diagnostics)?
         }
-        _ => function_call_expr(
-            loc,
-            ty,
-            args,
-            context,
-            ns,
-            symtable,
-            diagnostics,
-            resolve_to,
-        )?,
+        _ => {
+            if old_style_constructor_arguments(ty, diagnostics) {
```

Comment:
It would be more idiomatic to return an `Err(())` directly from the `old_style_constructor_arguments()` function. Which would allow for doing `old_style_constructor_arguments(ty, diagnostics)?;` instead.

Reply:
Yes, you're right, that would be more idiomatic.